### PR TITLE
add sympy and mpmath

### PR DESCRIPTION
**Pull request by msarahan**
@asmeurer - you might have some interest in these.  I have added you as maintainer.

### CODE REVIEW

File: recipes/mpmath/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    - mpmath
+    - mpmath.calculus
+    - mpmath.functions
+    - mpmath.libmp
+    - mpmath.matrices
+    - mpmath.tests
+
+about:
+  home: http://mpmath.org
+  license: BSD
```
Comment (jakirkham) [inline recipes/mpmath/meta.yaml]: Version?

Comment (jakirkham) [inline recipes/mpmath/meta.yaml]: Should we do `license_family` here too? I don't care either way, but figured we might as well given the other one hase it.

File: recipes/mpmath/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  version: {{ version }}
+
+source:
+  fn: mpmath-{{ version }}.tar.gz
+  url: https://pypi.python.org/packages/source/m/mpmath/mpmath-{{ version }}.tar.gz
+  md5: af5cc956b2673b33a25c3e57299bae7b
+
+build:
+  number: 0
+  script: python -m pip install --no-deps .
```
Comment (jakirkham) [inline recipes/mpmath/meta.yaml]: Still feel we should stick to the standard.

    python setup.py install --single-version-externally-managed --record record.txt

File: recipes/sympy/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  version: {{ version }}
+
+source:
+  fn: sympy-{{ version }}.tar.gz
+  url: https://pypi.python.org/packages/source/s/sympy/sympy-{{ version }}.tar.gz
+  md5: 43e797de799f00f9e8fd2307dba9fab1
+
+build:
+  number: 0
+  script: python -m pip install --no-deps .
```
Comment (jakirkham) [inline recipes/sympy/meta.yaml]: Ditto.

    python setup.py install --single-version-externally-managed --record record.txt

File: recipes/mpmath/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  summary: Python library for arbitrary-precision floating-point arithmetic
+  description: |
+    mpmath is a free Python library for real and complex floating-point
+    arithmetic with arbitrary precision.
+  doc_url: http://mpmath.org/doc/current/
+  dev_url: https://github.com/fredrik-johansson/mpmath
+
+extra:
+  recipe-maintainers:
+    - asmeurer
```
Comment (jakirkham) [inline recipes/mpmath/meta.yaml]: Happy to be a maintainer on this, @asmeurer?

File: recipes/sympy/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  description: |
+    SymPy is a Python library for symbolic mathematics. It aims to become a
+    full-featured computer algebra system (CAS) while keeping the code as
+    simple as possible in order to be comprehensible and easily extensible.
+  doc_url: http://docs.sympy.org/latest/index.html
+  dev_url: https://github.com/sympy/sympy
+
+extra:
+  recipe-maintainers:
+    - asmeurer
```
Comment (jakirkham) [inline recipes/sympy/meta.yaml]: Happy to be a maintainer on this, @asmeurer?

**Comment by jakirkham** [comment]
LGTM @msarahan. Going to wait for @asmeurer response to the above questions, but I fully expect the answer will be yes in both cases.

File: recipes/sympy/meta.yaml (hunk 5 above)
Comment (scopatz) [inline recipes/sympy/meta.yaml]: You can add me too as a maintainer.

Comment (jakirkham) [inline recipes/sympy/meta.yaml]: Thanks @scopatz.

Could you please give @asmeurer a nudge? 😉

Comment (scopatz) [inline recipes/sympy/meta.yaml]: Nudged, but I think he is pretty busy with the sympy paper.

Comment (jakirkham) [inline recipes/sympy/meta.yaml]: Thanks. It's ok we can be patient.

**Comment by asmeurer** [comment]
Yeah, I'll do it. 

Please make SymPy depend on fastcache. It's an optional dependency, but there's no reason to not install it if you're using conda.

**Comment by asmeurer** [comment]
gmpy would also be a nice optional dependency to toss in for both sympy and mpmath, once it is built.

File: recipes/sympy/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  number: 0
+  script: python setup.py install
+
+requirements:
+  build:
+    - python
+    - mpmath
+  run:
+    - python
+    - mpmath >=0.19
```
Comment (jakirkham) [inline recipes/sympy/meta.yaml]: Also, please add `fastcache` here too.

**Comment by jakirkham** [comment]
Minor comments based on @asmeurer's feedback and then I think we are good to go. Don't think we have `gmpy` yet.

**Comment by jakirkham** [comment]
Thanks @msarahan.

**Pull request merged**